Sources/Tracing/TracingTime.swift

```diff
@@ -0,0 +1,72 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Distributed Tracing open source project
+//
+// Copyright (c) 2020-2022 Apple Inc. and the Swift Distributed Tracing project
+// authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if swift(>=5.6.0)
+@_implementationOnly @preconcurrency import struct Dispatch.DispatchWallTime
+#else
+@_implementationOnly import struct Dispatch.DispatchWallTime
+#endif
+
+/// A wall-clock based time value used to mark the beginning and end of a trace ``Span``.
+///
+/// ### Rationale for this type
+/// This type is introduced in order to abstract between various time sources, such as
+/// Dispatch's `DispatchWallTime` or any potential future `Clock` type in the Swift
+/// standard library (or elsewhere) that we might want to use as source of time.
+///
+/// Abstracting away the source of the measurement into this wrapper allows us to
+/// have ``Tracer`` implementations don't care about the user API that is surfaced as `startSpan`
+/// to end-users. As time goes on, this API may evolve, but we will not have to make breaking changes
+/// to tracer implementations themselves.
+public struct TracingTime: Sendable {
+  enum Repr {
+      case dispatchWallTime(DispatchWallTime)
+  }
+  private var repr: Repr
+
+  private init(dispatchWallTime: DispatchWallTime) {
```

Comment:
why not public?

Reply:
i attempted to hide away dispatch completely, debatable if we care to do so or not